libraries/node-core-library/src/LockFile.ts

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fsx from 'fs-extra';
+import * as path from 'path';
+import * as child_process from 'child_process';
+
+export function getProcessStartTime(pid: string): string | undefined {
+  let args: string[];
+  if (process.platform === 'darwin') {
+    args = [`-p ${pid}`, '-f'];
+  } else if (process.platform === 'linux') {
+    args = ['-p', pid.toString(), '-f'];
+  } else {
+    throw new Error(`Unsupported system: ${process.platform}`);
+  }
+
+  const psResult: string = child_process.spawnSync('ps', args).stdout.toString();
+  const psSplit: string[] = psResult.split('\n');
+
+  if (psSplit[1]) {
+    // note that on OSX the first entry is a ""
+    // UID, PID, PPID, C, STIME, TTY, TIME, CMD
+    // e.g.: items = ['root', '2848', '1', '0', '13:08', '?', '00:00:00', '/usr/lib/udisks2/udisksd']
```

Comment:
Different "ps" implementations will have different behaviors and formatting.  Can we make this check more robust?